examples/applications/plot_cyclical_feature_engineering.py

```diff
@@ -104,7 +104,13 @@
 # train machine learning models with cross validation. Instead, we simplify the
 # representation by collapsing those into the `"rain"` category.
 #
-X["weather"].replace(to_replace="heavy_rain", value="rain", inplace=True)
+X["weather"] = (
+    X["weather"]
+    .astype(object)
+    .replace(to_replace="heavy_rain", value="rain")
+    .astype("category")
+)
```

Comment:
Or

```suggestion
X.loc[X["weather"] == "heavy_rain", "col"] = "rain"
X["weather"] = X["weather"].astype("category").cat.remove_unused_categories()
```

but don't know which one's better

Reply:
I don't like mine, but I haven't found anything better after looking for 20 minutes on how to merge two categorical values ...

I find the flow is more readable somehow than your suggestion. I agree it is not super clear why you need `.astype(object)` and then `.astype('category')` (avoid yet another pandas warning, `replace` is not supposed to change categories).

Reply:
Actually I found this which maybe is a bit simpler?
```py
X["weather"] = X["weather"].cat.remove_categories("heavy_rain").fillna("rain")
```

Reply:
if there are already missing values, this removes them though. So I don't think we'd want to encourage the pattern in our examples.

Reply:
Hmmm good point actually ... I reverted this commit and went with my first proposal